lib/Runtime/Debug/TTActionEvents.cpp

```diff
@@ -361,32 +361,24 @@ namespace TTD
         void AllocateFunctionAction_Execute(const EventLogEntry* evt, ThreadContextTTD* executeContext)
         {
             TTD_REPLAY_ACTIVE_CONTEXT(executeContext);
-            const JsRTSingleVarScalarArgumentAction* action = GetInlineEventDataAs<JsRTSingleVarScalarArgumentAction, EventKind::AllocateFunctionActionTag>(evt);
+            const JsRTSingleVarArgumentAction* action = GetInlineEventDataAs<JsRTSingleVarArgumentAction, EventKind::AllocateFunctionActionTag>(evt);
 
-            Js::Var res = nullptr;
-            if(!GetScalarItem_0(action))
+            Js::Var nameVar = InflateVarInReplay(executeContext, GetVarItem_0(action));
+
+            Js::JavascriptString* name = nullptr;
+            if(nameVar != nullptr)
```

Comment:
For a closer match with the non-TTD code, should this check `nameVar != JS_INVALID_REFERENCE`? And maybe also call it `metadata`, even though it is currently just a name?

Reply:
JS_INVALID_REFERENCE isn't used internally, so I didn't make that change, but I did update the variable to metadata.